tasks/rulegen/src/main.rs

```diff
@@ -175,6 +175,39 @@ fn format_code_snippet(code: &str) -> String {
     format!("r#\"{}\"#", code.replace("\\\"", "\""))
 }
 
+fn dedent(s: &str) -> String {
```

Comment:
would you mind adding tests for this method? we should add as many as we can from this file probably: https://github.com/dmnd/dedent/blob/main/src/dedent.test.ts.

I don't think it's necessary that we pass all of them (even adding them to the code but commented out is fine), but we should try to pass as many as possible

Reply:
I have added tests for dedent and all the tests pass. Please help me take a look!
https://github.com/oxc-project/oxc/pull/6866/files#diff-3326aec4ab83a43d557ded8a12030db8cb2028e045caa2551ac5c78ae2a18a40